### PR TITLE
[MRG+1] Emit warnings / custom errors for differencing issues

### PR DESCRIPTION
**Pull request by tgsmith61591**
# Description

Two recent issues pointed out areas where a large enough `m` could seasonally difference an array into being too small for fitting downstream models. This PR adds a custom error for #341 and a warning for #351 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation change

# How Has This Been Tested?

Added new tests that capture the warning / custom error and assert on them.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pmdarima/arima/tests/test_auto.py [hunk 1]
```diff
@@ -17,3 +17,36 @@ def test_deprecation_warnings_on_class():
...
+            D=1,
+            trace=True,
+            error_action='ignore',
+            suppress_warnings=True,
+            stepwise=True
+        )
+
+    # assert that we catch the np LinAlg error and reraise with a more
+    # meaningful message
+    except ValueError as v:
```
Comment (aaronreidsmith) [inline pmdarima/arima/tests/test_auto.py:49]: I think the [preferred way](https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions) to do this in pytest is like this:

```python
with pytest.raises(ValueError) as error:
    auto.auto_arima(
        y,
        start_p=1,
        start_q=1,
        test='adf',
        max_p=3,
        max_q=3,
        m=52,
        start_P=0,
        seasonal=True,
        d=None,
        D=1,
        trace=True,
        error_action='ignore',
        suppress_warnings=True,
        stepwise=True
    )

assert "Encountered exception in stationarity test" in str(error.value)
```

Comment (aaronreidsmith) [reply pmdarima/arima/tests/test_auto.py:49]: Or I guess you could use `pmdarima.compat.pytest.pytest_error_str` as we do in `test_utils.py`

File: pmdarima/tests/test_context_managers.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    try:
+        with ctx.except_and_reraise(
+                ValueError,
+                raise_err=KeyError,
+                raise_msg="bar message"
+        ):
+            raise ValueError("contains foo message")
+    except KeyError as k:
+        assert "bar message" in str(k)
+        assert "raised from ValueError" in str(k)
```
Comment (aaronreidsmith) [inline pmdarima/tests/test_context_managers.py:19]: Same as my other comment

File: pmdarima/arima/tests/test_auto.py (hunk 1 above)
Comment (tgsmith61591) [reply pmdarima/arima/tests/test_auto.py:49]: Yes, we should. Not sure why I didn't do that here.

**Pull request merged**